packages/integrations/prefetch/src/client.ts

```diff
@@ -53,7 +54,14 @@ async function preloadHref(link: HTMLAnchorElement) {
 		const html = parser.parseFromString(contents, 'text/html');
 		const styles = Array.from(html.querySelectorAll<HTMLLinkElement>('link[rel="stylesheet"]'));
 
-		await Promise.all(styles.map((el) => fetch(el.href)));
+		await Promise.all(
+			styles
+				.filter((el) => !loadedStyles.has(el.href))
+				.map((el) => {
+					loadedStyles.add(el.href);
+					return fetch(el.href);
```

Comment:
This promise needed to be returned, which might fix the tests! Watching CI to see.

Reply:
@Jelenkee can you try this?